### ISSUE TITLE
Access Location header of successful upload?

### ISSUE THREAD

**Issue by lougreenwood** [tim-evans/ember-file-upload#156]
My API returns a 201 response code with the `location` header containing the URL of the uploaded file.

I see in the developer console network monitor that the `Location` header is included in the network response, but the `headers` object which `uploadBinary()` returns does not contain this header, instead it only has 3 headers:

* cache-control
* expires
* pragma

Is this intentional? I see that any 2XX response should be handled as a resolved upload, so it seems that `Location` should probably be included to allow following the IETF spec: 

> The 201 (Created) status code indicates that the request has been fulfilled and has resulted in one or more new resources being created.  The primary resource created by the request is identified by either a Location header field in the response or, if no Location field is received, by the effective request URI.
https://tools.ietf.org/html/rfc7231#section-6.3.2

**Comment by lougreenwood** [comment]
Seems that according to the readme, `Location` should actually be supported. here's the response I get from `uploadBinary()`:

`{status: 201, body: "", headers: {cache-control: "no-cache, no-store, must-revalidate", expires: "0", pragma: "no-cache"}}`

Seems that there are possibly a few things different about my setup which could cause this bug:

* return 201 status code (add-on tests only check for 200, but code allows any 2xx)
* empty body
* use of `uploadBinary()` instead of plain `upload()` in the README

**Comment by lougreenwood** [comment]
After some investigation, it became apparent that this was an issue with our CORS setup. (I noticed that `XMLHttpRequest.getAllResponseHeaders()` was a culprit for returning only those limited headers).

Seems that we needed to add `Access-Control-Expose-Headers: Location` to our server response:
https://stackoverflow.com/questions/8945587/xmlhttprequest-getallresponseheaders-not-returning-all-the-headers

Now receiving `Location in `uploadBinary()` response.

False alarm! 😃
